web/packages/teleport/src/Discover/SelectResource/getMarketingMatches.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright 2023 Gravitational, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  ClusterResource,
+  MarketingParams,
+} from 'teleport/services/userPreferences/types';
+
+/**
+ * Returns a list of resources that match provided marketing parameters.
+ *
+ * @param marketingParams - MarketingParams from cluster user preferences which are set at signup
+ * @returns an array of ClusterResources associated with the marketing params for resource discoverability
+ *
+ */
+export const getMarketingMatches = (
+  marketingParams: MarketingParams
+): ClusterResource[] => {
+  const params = [];
+  if (marketingParams) {
+    marketingParams.campaign && params.push(marketingParams.campaign);
+    marketingParams.medium && params.push(marketingParams.medium);
+    marketingParams.source && params.push(marketingParams.source);
+    marketingParams.intent && params.push(marketingParams.intent);
+  }
+  if (params.length === 0) {
+    return [];
+  }
+
+  const matches = [];
+  params.forEach(p => {
+    Object.values(TermMatches).forEach(m => {
+      const clusterResource = matchLookup(m);
+      if (
+        p.includes(m) &&
+        clusterResource &&
+        !matches.includes(clusterResource)
+      ) {
+        matches.push(clusterResource);
+      }
+    });
+  });
+
+  return matches;
+};
+
+export enum TermMatches {
```

Comment:
```suggestion
export enum TermMatch {
```
Additionally, enum values in TS are usually written in camel case: `App`, `Database` etc.